### ISSUE TITLE
Crash on SI-5421 example

### ISSUE THREAD

**Issue by lihaoyi** [lihaoyi/Ammonite#51]
```scala
@ class A {def t: Int = {println("getter"); -282}; def t_=(p: Int) { println("setter") }; def m(v: String) = new A}
defined class A
@ val a = new A
a: cmd19.A = cmd19$A@4641e3dd
@ a m "txt" t = 18; println("x")
java.util.NoSuchElementException: Either.left.value on Right
	scala.util.Either$LeftProjection.get(Either.scala:289)
	ammonite.repl.interp.Preprocessor$$anon$2.apply(Preprocessor.scala:91)
	ammonite.repl.interp.Interpreter$$anonfun$processLine$2.apply(Interpreter.scala:36)
	ammonite.repl.interp.Interpreter$$anonfun$processLine$2.apply(Interpreter.scala:32)
	ammonite.repl.Catching.flatMap(Util.scala:73)
	ammonite.repl.interp.Interpreter.processLine(Interpreter.scala:32)
	ammonite.repl.Repl$$anonfun$action$1$$anonfun$apply$4.apply(Repl.scala:44)
	ammonite.repl.Repl$$anonfun$action$1$$anonfun$apply$4.apply(Repl.scala:43)
	ammonite.repl.frontend.Scoped$$anonfun$flatMap$1.apply(Signaller.scala:43)
	ammonite.repl.frontend.Signaller.apply(Signaller.scala:28)
	ammonite.repl.frontend.Scoped$class.flatMap(Signaller.scala:43)
	ammonite.repl.frontend.Signaller.flatMap(Signaller.scala:10)
	ammonite.repl.Repl$$anonfun$action$1.apply(Repl.scala:43)
	ammonite.repl.Repl$$anonfun$action$1.apply(Repl.scala:42)
	ammonite.repl.Res$Success.flatMap(Util.scala:22)
	ammonite.repl.Repl.action(Repl.scala:42)
	ammonite.repl.Repl.loop$1(Repl.scala:53)
	ammonite.repl.Repl.run(Repl.scala:56)
	ammonite.repl.Repl$.main(Repl.scala:80)
	$line1.$read$$iw$$iw$.<init>(<console>:8)
	$line1.$read$$iw$$iw$.<clinit>(<console>)
	$line1.$eval$.$print$lzycompute(<console>:7)
	$line1.$eval$.$print(<console>:6)
	$line1.$eval.$print(<console>)
	sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:483)
	scala.tools.nsc.interpreter.IMain$ReadEvalPrint.call(IMain.scala:755)
	scala.tools.nsc.interpreter.IMain$Request.loadAndRun(IMain.scala:1002)
	scala.tools.nsc.interpreter.IMain$WrappedRequest$$anonfun$loadAndRunReq$1.apply(IMain.scala:609)
	scala.tools.nsc.interpreter.IMain$WrappedRequest$$anonfun$loadAndRunReq$1.apply(IMain.scala:608)
	scala.reflect.internal.util.ScalaClassLoader$class.asContext(ScalaClassLoader.scala:31)
	scala.reflect.internal.util.AbstractFileClassLoader.asContext(AbstractFileClassLoader.scala:19)
	scala.tools.nsc.interpreter.IMain$WrappedRequest.loadAndRunReq(IMain.scala:608)
	scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:540)
	scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:536)
	xsbt.ConsoleInterface$$anon$2.createInterpreter(ConsoleInterface.scala:54)
	scala.tools.nsc.interpreter.ILoop$$anonfun$process$1.apply$mcZ$sp(ILoop.scala:852)
	scala.tools.nsc.interpreter.ILoop$$anonfun$process$1.apply(ILoop.scala:850)
	scala.tools.nsc.interpreter.ILoop$$anonfun$process$1.apply(ILoop.scala:850)
	scala.reflect.internal.util.ScalaClassLoader$.savingContextLoader(ScalaClassLoader.scala:97)
	scala.tools.nsc.interpreter.ILoop.process(ILoop.scala:850)
	scala.tools.nsc.interpreter.ILoop.main(ILoop.scala:872)
	xsbt.ConsoleInterface.run(ConsoleInterface.scala:62)
	sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:483)
	sbt.compiler.AnalyzingCompiler.call(AnalyzingCompiler.scala:101)
	sbt.compiler.AnalyzingCompiler.console(AnalyzingCompiler.scala:76)
	sbt.Console.sbt$Console$$console0$1(Console.scala:22)
	sbt.Console$$anonfun$apply$2$$anonfun$apply$1.apply$mcV$sp(Console.scala:23)
	sbt.Console$$anonfun$apply$2$$anonfun$apply$1.apply(Console.scala:23)
	sbt.Console$$anonfun$apply$2$$anonfun$apply$1.apply(Console.scala:23)
	sbt.Logger$$anon$4.apply(Logger.scala:85)
	sbt.TrapExit$App.run(TrapExit.scala:248)
	java.lang.Thread.run(Thread.java:745)
Something unexpected went wrong =(
```

**Comment by lihaoyi** [comment]
Looks like a bug in the parser

```scala
@ new scalaParser.Scala("""{a.m("txt").t = 18}""").TmplBody.run()
res3: scala.util.Try[Unit] = Success(())
@ new scalaParser.Scala("""{a m "txt" t = 18}""").TmplBody.run()
res4: scala.util.Try[Unit] = Failure(ParseError(Position(13,1,14), Position(13,1,14), <24 traces>))
```

**Comment by lihaoyi** [comment]
Fixed by 4826fdbc945933dc375d71d01662ab7daf349b07
